lib/matplotlib/backends/qt_compat.py

```diff
@@ -216,3 +220,57 @@ def _setDevicePixelRatioF(obj, val):
     elif hasattr(obj, 'setDevicePixelRatio'):
         # Not available on Qt4 or some older Qt5.
         obj.setDevicePixelRatio(val)
+
+
+class _allow_interrupt:
+    def __init__(self, qApp):
+        self.interrupted_qobject = qApp
+        self.old_fd = None
+        self.old_sigint_handler = None
+        self.interrupt_caught = None
+
+        QAS = QtNetwork.QAbstractSocket
+        self.qt_socket = QAS(QAS.TcpSocket, qApp)
+        # Create a socket pair
+        self.wsock, self.rsock = socket.socketpair()
+        # Let Qt listen on the one end
+        self.qt_socket.setSocketDescriptor(self.rsock.fileno())
+        self.wsock.setblocking(False)
+        # First Python code executed gets any exception from
+        # the signal handler, so add a dummy handler first
+        self.qt_socket.readyRead.connect(lambda: None)
```

Comment:
I don't think we actually need this.  Because we are installing our own signal handler the `KeyboardInterupt` does not get injected so we don't have this problem.